resources/helm/overlays/istio_cni/templates/daemonset-v2.3.yaml

```diff
@@ -0,0 +1,128 @@
+{{ if .Values.cni.enabled }}
+{{- if and (.Values.cni.supportedReleases) (eq .Values.cni.instanceVersion "v2.3") }}
+# This manifest installs the Istio install-cni container, as well
+# as the Istio CNI plugin and config on
+# each master and worker node in a Kubernetes cluster.
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: istio-cni-node-v2.3
+  namespace: {{ .Release.Namespace }}
+  labels:
+    k8s-app: istio-cni-node-v2-3
+    release: {{ .Release.Name }}
+spec:
+  selector:
+    matchLabels:
+      k8s-app: istio-cni-node-v2-3
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 1
+  template:
+    metadata:
+      labels:
+        k8s-app: istio-cni-node-v2-3
+        release: {{ .Release.Name }}
+      annotations:
+        sidecar.istio.io/inject: "false"
+        # Custom annotations
+        {{- if .Values.cni.podAnnotations }}
+{{ toYaml .Values.cni.podAnnotations | indent 8 }}
+        {{- end }}
+    spec:
+      nodeSelector:
+        kubernetes.io/os: linux
+      tolerations:
+        # Make sure istio-cni-node gets scheduled on all nodes.
+        - effect: NoSchedule
+          operator: Exists
+        # Mark the pod as a critical add-on for rescheduling.
+        - key: CriticalAddonsOnly
+          operator: Exists
+        - effect: NoExecute
+          operator: Exists
+      priorityClassName: system-node-critical
+      serviceAccountName: istio-cni
+      # Minimize downtime during a rolling upgrade or deletion; tell Kubernetes to do a "force
+      # deletion": https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods.
+      terminationGracePeriodSeconds: 5
+{{- if .Values.cni.imagePullSecrets }}
+      imagePullSecrets:
+{{- range .Values.cni.imagePullSecrets }}
+      - name: {{ . }}
+{{- end }}
+{{- end }}
+      containers:
+        # This container installs the Istio CNI binaries
+        # and CNI network config file on each node.
+        - name: install-cni-v2-3
+          image: "{{ .Values.cni.image_v2_3 }}"
+          imagePullPolicy: {{ .Values.cni.imagePullPolicy }}
+          livenessProbe:
+            httpGet:
+              path: /healthz
+              port: 8000
+          initialDelaySeconds: 5
+          readinessProbe:
+            httpGet:
+              path: /readyz
+              port: 8000
+          command: ["install-cni"]
+          securityContext:
+            privileged: true
+          env:
+{{- if .Values.cni.cniConfFileName_v2_3 }}
+            # Name of the CNI config file to create.
+            - name: CNI_CONF_NAME
+              value: "{{ .Values.cni.cniConfFileName_v2_3 }}"
+{{- end }}
+            # The CNI network config to install on each node.
+            - name: CNI_NETWORK_CONFIG
+              valueFrom:
+                configMapKeyRef:
+                  name: istio-cni-config
```

Comment:
I'd prefer to see the `istio-cni-configmap` also split into two ConfigMaps, so that every version above 2.3 has its own ConfigMap. This should make it easier to rebase to new versions.